crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2426,6 +2426,40 @@ impl<'db> Type<'db> {
         })
     }
 
+    fn enter(self, db: &'db dyn Db) -> Type<'db> {
+        self.try_enter(db)
+            .unwrap_or_else(|err| err.fallback_enter_type(db))
+    }
+
```

Comment:
We should be quite careful about methods like these, because it means the path of least resistance for a caller is to ignore errors entirely. If we provide this, it should have a warning doc comment, like `Type::bool` and `Type::iterate` have.